### PR TITLE
Fix sdk/template live tests

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
I don't have all the context but the changes look reasonable to me.

**Comment by heaths** [comment]
> I don't have all the context but the changes look reasonable to me.

@weshaggard thanks. Basically, the _autorest.md_ used the wrong scope, but this also points to a potential problem: Key Vault requires a separate policy, and DPG doesn't seem to allow customizing the pipeline (at least, not obviously). It just so happens that the Core-provided `BearerTokenAuthenticationPolicy` works, but not in all cases. So I'm able to work around the live test issues, but if I were doing this for a real client, I would need to potentially customize pipeline policies.

/cc @chunyu3 @annelo-msft

File: sdk/template/Azure.Template/README.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ The `GetSecret` method retrieves a secret from the service.
 
 ```C# Snippet:Azure_Template_GetSecret
 string endpoint = "https://myvault.vault.azure.net";
+var credential = new DefaultAzureCredential();
```
[the remark this answers is not in the document]
Comment (heaths) [reply sdk/template/Azure.Template/README.md:70]: Good catch. I'll fix that in my DPG update for sdk/template.

**Pull request merged**